rslib/src/search/parser.rs

```diff
@@ -472,6 +415,43 @@ fn parse_prop(s: &str) -> ParseResult<SearchNode> {
     })
 }
 
+fn parse_prop_float<'a, N>(num: &str, prop_clause: &'a str) -> ParseResult<'a, N>
+where
+    N: FromStr,
+    <N as FromStr>::Err: PartialEq<ParseFloatError>,
+{
+    num.parse()
+        .map_err(|_e| parse_failure(prop_clause, FailKind::InvalidNumber(prop_clause.to_owned())))
+}
+
+fn parse_prop_integer<'a, N>(num: &str, prop_clause: &'a str) -> ParseResult<'a, N>
+where
+    N: FromStr,
+    <N as FromStr>::Err: PartialEq<ParseIntError>,
+{
+    num.parse()
+        .map_err(|_e| parse_failure(prop_clause, FailKind::InvalidNumber(prop_clause.to_owned())))
+}
+
+fn parse_prop_rated<'a>(num: &str, prop_clause: &'a str) -> ParseResult<'a, PropertyKind> {
+    let mut it = num.splitn(2, ':');
+    let days = parse_prop_integer::<i32>(it.next().unwrap(), prop_clause)?.min(0);
+    let ease = match it.next() {
+        Some(v) => match parse_prop_integer(v, prop_clause)? {
+            u @ 1..=5 => EaseKind::AnswerButton(u),
+            _ => {
+                return Err(parse_failure(
+                    prop_clause,
+                    FailKind::InvalidRatedEase(prop_clause.to_owned()),
```

Comment:
If we wanted to go one step further, even this case could be covered by "invalid number provided".